### PR TITLE
Add support for Easun SV serie

### PR DESCRIPTION
**Pull request by sand1812**
Added support for Easun SV inverters.
Has been tested on SV-IV 5.6 with parallel boards.

All read requests works.
Some set requests return NACK
Some set requests not yet implemented (waiting to find out why some don't work)

### CODE REVIEW

File: mppsolar/protocols/pi18sv.py [hunk 1]
```diff
@@ -0,0 +1,1008 @@
...
+
+class pi18sv(AbstractProtocol):
+    def __str__(self):
+        return "PI18-SV protocol handler"
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__()
+        self._protocol_id = b"PI18SV"
+        self.COMMANDS = COMMANDS
+        self.STATUS_COMMANDS = ["PI","T","ET","EY","EM","ED","ID","VFW","PIRI","GS","MOD","FWS","FLAG","DI","CHGCR","MUCHGCR","PRI","PGS","ACCT","ACLT"]
```
Comment (jblance) [inline mppsolar/protocols/pi18sv.py:881]: STATUS_COMMANDS are the commands to run when a --getStatus command is issued. This list looks too long (normally it is 1 or 2 or 3 key commands)

File: mppsolar/protocols/pi18sv.py [hunk 2]
```diff
@@ -0,0 +1,1008 @@
...
+        self.SETTINGS_COMMANDS = [
```
Comment (jblance) [inline mppsolar/protocols/pi18sv.py:882]: as above, this is the --getSettings command, so the list is likely too long

**Review by jblance — CHANGES_REQUESTED** [review]
I've made a few comments, do they make sense?
Mostly the requested changes are to make the code easy to keep up to date for multiple different systems.

Thoughts?

**Comment by sand1812** [comment]
Hi,

Your comments looks right. I've lot of work today & tomorrow. I’ll update my code later this week. But feel free to do it if you want !

Do you have any idea why the SET commands are mostly not working ? 
For example, according to doc, « BUCD500,540 » should be ok, but the inverter returns 0 (failed)

I attached the doc from easun for reference.

Kind regards,

Greg.

**Comment by jblance** [comment]
sorry i didnt see the doc
re commands not working, first check with debug to make sure the correct command is being sent - commas in particular are a bit half removed
the other thing ive seen is different padding for values, sometimes extra leading seros are needed

**Pull request merged**